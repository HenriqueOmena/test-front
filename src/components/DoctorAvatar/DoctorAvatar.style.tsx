import { makeStyles } from "@material-ui/core";

const DoctorAvatarStyle = makeStyles(() => ({
  avatar: {
    margin: 15,
    minWidth: 200,
    display: "flex",
    alignItems: "center",
    flexDirection: "column"
  }
}));

export default DoctorAvatarStyle;
