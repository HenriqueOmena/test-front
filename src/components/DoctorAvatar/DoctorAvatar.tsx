import { Avatar } from "@material-ui/core";
import React from "react";
import DoctorAvatarStyle from "./DoctorAvatar.style";

const DoctorAvatar: React.FC<{ name: string }> = props => {
  const classes = DoctorAvatarStyle();
  return (
    <div className={classes.avatar}>
      <Avatar alt={props.name} />
      {props.name}
    </div>
  );
};

export default DoctorAvatar;
