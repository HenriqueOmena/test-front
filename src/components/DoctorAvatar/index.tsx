export { default as DoctorAvatar } from "./DoctorAvatar";
