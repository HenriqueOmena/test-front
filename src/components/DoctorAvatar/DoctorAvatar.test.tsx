import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import DoctorAvatar from "./DoctorAvatar";

describe("Component - DoctorAvatar", () => {
  const propMock = {
    name: "someName"
  };

  let component = <DoctorAvatar {...propMock} />;

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render with prop name", async () => {
    const { getByText } = render(component);
    const nameText = getByText(propMock.name);

    expect(nameText).toBeVisible();
  });
});
