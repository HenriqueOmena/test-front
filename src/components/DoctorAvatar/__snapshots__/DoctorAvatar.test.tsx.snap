// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DoctorAvatar Should Match SnapShot 1`] = `
<DoctorAvatar
  name="someName"
/>
`;
